src/code/z_camera.c

```diff
@@ -55,31 +256,184 @@ s32 Camera_IsSwimming(Camera* camera) {
+        return ((Player*)focalActor)->stateFlags2 & PLAYER_STATE2_800;
+    } else {
+        return 0;
+    }
+}
+
+s32 Camera_IsPlayerFormZora(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->transformation == PLAYER_FORM_ZORA;
+    } else {
+        return false;
+    }
+}
+
+s32 func_800CB924(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->stateFlags3 & PLAYER_STATE3_1000;
+    } else {
+        return 0;
+    }
+}
+
+s32 func_800CB950(Camera* camera) {
+    Player* player;
+    s32 phi_v0;
+    s32 ret;
+    f32 yDiff;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        yDiff = Camera_fabsf(camera->focalActorPosRot.pos.y - camera->playerFloorHeight);
+
+        phi_v0 = false;
+        if (yDiff < 11.0f) {
+            phi_v0 = true;
+        }
+
+        ret = phi_v0;
+
+        if (!ret) {
+
+            ret = false;
+
+            if (camera->focalActor->gravity > -0.1f) {
+                ret = true;
+            }
+
+            player = (Player*)camera->focalActor;
+            if (!ret) {
+                // Using zora fins
+                ret = player->stateFlags1 & PLAYER_STATE1_200000;
+                ret = !!ret;
+            }
+        }
+        return ret;
+    } else {
+        return true;
+    }
+}
+
+s32 Camera_IsClimbingLedge(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->stateFlags1 & PLAYER_STATE1_4;
+    } else {
+        return 0;
+    }
+}
+
+s32 Camera_IsChargingSwordOrDekuFlowerDive(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+    s32 ret;
+
+    if (focalActor == &GET_PLAYER(camera->play)->actor) {
+        // Charging Sword
+        ret = !!(((Player*)focalActor)->stateFlags1 & PLAYER_STATE1_1000);
+        if (!ret) {
+            // Deku Flower Dive
+            ret = !!(((Player*)focalActor)->stateFlags3 & PLAYER_STATE3_100);
+        }
+        return ret;
+    } else {
+        return false;
+    }
+}
+
+s32 func_800CBA7C(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->stateFlags2 & PLAYER_STATE2_800000;
+    } else {
+        return 0;
+    }
+}
+
+s32 func_800CBAAC(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB8F4.s")
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->meleeWeaponState;
+    } else {
+        return 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB924.s")
+s32 func_800CBAD4(Vec3f* dst, Camera* camera) {
```

Comment:
So is this just GetFocalActorPos?

Reply:
What is `bodyPartsPos[0]` from player?

Reply:
It should be this
```
/* 0x00 */ PLAYER_BODYPART_WAIST, // PLAYER_LIMB_WAIST
```
But to be honest I grabbed that enum from OoT and haven't checked if the values are the same on MM. 
There's no reason to think those values changed between the two games, specially when the limb enum didn't change, but you never know...